### PR TITLE
feat(cluster): support compression for query flight data

### LINKED ISSUE #13885: feat: compression when data shuffle via flight RPC
**Summary**

When Databend shuffles data between cluster nodes via Flight RPC, network bandwidth can become a critical issue.

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat(cluster): support compression for query flight data

- Closes #13885

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13934)

### CODE REVIEW

File: src/query/settings/src/settings_default.rs [hunk 1]
```diff
@@ -531,6 +531,12 @@ impl DefaultSettings {
                     possible_values: None,
                     mode: SettingMode::Both,
                 }),
+                ("query_flight_compression", DefaultSettingValue {
+                    value: UserSettingValue::String(String::from("None")),
+                    desc: "flight compression method",
+                    possible_values: Some(vec!["None", "Lz4"]),
```
Comment (sundy-li) [inline src/query/settings/src/settings_default.rs:537]: zstd

File: src/query/settings/src/settings_default.rs [hunk 2]
```diff
@@ -531,6 +531,12 @@ impl DefaultSettings {
                     possible_values: None,
                     mode: SettingMode::Both,
                 }),
+                ("query_flight_compression", DefaultSettingValue {
+                    value: UserSettingValue::String(String::from("Lz4")),
+                    desc: "flight compression method",
+                    possible_values: Some(vec!["None", "Lz4", "Zstd"]),
+                    mode: SettingMode::Both,
+                }),
```
Comment (BohuTANG) [inline src/query/settings/src/settings_default.rs:539]:
```suggestion
                ("query_flight_compression", DefaultSettingValue {
                    value: UserSettingValue::String(String::from("LZ4")),
                    desc: "flight compression method",
                    possible_values: Some(vec!["None", "LZ4", "ZSTD"]),
                    mode: SettingMode::Both,
                }),
```

**Review by sundy-li — APPROVED** [review]

File: src/query/settings/src/settings_getter_setter.rs [hunk 3]
```diff
@@ -24,6 +24,12 @@ use crate::ReplaceIntoShuffleStrategy;
 use crate::ScopeLevel;
 use crate::SettingMode;
 
+#[derive(Clone, Copy)]
+pub enum FlightCompression {
+    Lz4,
+    Zstd,
+}
```
Comment (BohuTANG) [inline src/query/settings/src/settings_getter_setter.rs:31]:
```suggestion
pub enum FlightCompression {
    LZ4,
    ZSTD,
}
```

Comment (BohuTANG) [reply src/query/settings/src/settings_getter_setter.rs:31]: This display will the enum?
image

File: src/query/settings/src/settings_default.rs [hunk 4]
```diff
@@ -531,6 +531,12 @@ impl DefaultSettings {
                     possible_values: None,
                     mode: SettingMode::Both,
                 }),
+                ("query_flight_compression", DefaultSettingValue {
+                    value: UserSettingValue::String(String::from("None")),
```
Comment (BohuTANG) [inline src/query/settings/src/settings_default.rs:535]: From my test using this image(https://github.com/datafuselabs/databend/pull/13934#issuecomment-1844108465), `query_flight_compression` default value is Lz4 not None, weird.

File: src/query/settings/src/settings_getter_setter.rs (hunk 3 above)
Comment (BohuTANG) [reply src/query/settings/src/settings_getter_setter.rs:31]: From the settings show, user may set as:
```
set query_flight_compression='Lz4';
```
not 
```
set query_flight_compression='LZ4';
```

**Pull request merged**